README.md

```diff
@@ -76,6 +76,44 @@ func main() {
 }
 ```
 
+
+## Event Broker
+by default, this module executes commands in synchronous mode (ie: send Command and wait Response). 
+AMI connections are asynchronous.
+You can enable asynchronous mode adding next code just before the Login Command
+
+```Go
+	// Start Event Broker
+	socket.Events().Start(ctx, socket)
+```
+Now, you can subscribe for events:
+
+```Go
+	events := h.socket.Events().Subscribe()
+	defer func() {
+		h.socket.Events().Unsubscribe <- events
+	}()
+
+	for {
+		select {
+		//canceled
+		case <-ctx.Done():
+			return
+		// received event from asterisk
+		case response := <-events:
+			js, _ := json.Marshal(response)
+			fmt.Printf(w, "data: %s\n\n", js)
+		}
+	}
+
+```
+### Advantages of using Event Broker
+* Multiple subscribers (ie: http request por events)
+* The Action ID from response is compared width the action ID from command, so you have accurate response
```

Comment:
Agree with pros, but the event broker IMHO should be implemented at the top of library, not inside it.
For example:
 https://github.com/heltonmarx/goami/blob/master/example/asterisk.go#L90

Consuming the ami.Events(ctx, socket) by a dedicated socket. 
And using another socket connection to submit commands only.

Creating a collection of sockets + events dedicated to your application, but  not necessarily included to the library.

Reply:
the main reason why I put the event broker inside the library was because of the risk of a command "stealing" the response belonging to another command, especially in multi-client environments (rest api).
Let me figure how to solve that problem without including an event broker.